hw_4 numpy/numpy_challenge.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+
+
+if __name__ == '__main__':
+    first = np.full((14, 14), 0)
+    second = np.mgrid[1:7, 1:7]
+    third = np.zeros((14, 14), int)
+    np.fill_diagonal(third, 4)
+
+
+
+def matrix_multiplication(mat1, mat2):
+    return np.matmul(mat1, mat2)
+
+def multiplication_check(list_of_matrices):
+    rows, cols = [], []
+    for mat in list_of_matrices:
+        row, col = mat.shape
+        rows.append(row)
+        cols.append(col)
+    if rows[1:] == cols[:-1]:
+        return True
+    else:
+        return False
```

Comment:
Не самый интуитивный способ. Последнее условие можно упростить
```suggestion
    rows, cols = [], []
    for mat in list_of_matrices:
        row, col = mat.shape
        rows.append(row)
        cols.append(col)
    return rows[1:] == cols[:-1]
```